src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,83 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import React from 'react';
+
+interface PaginationProps {
+  total: number;
+  perPage: number;
+  currentPage?: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<PaginationProps> = ({
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const handlePageChange = (page: number) => {
+    if (page !== currentPage && page >= 1 && page <= totalPages) {
+      onPageChange(page);
+    }
+  };
+
+  return (
+    <ul className="pagination">
+      <li className={classNames('page-item', { disabled: currentPage === 1 })}>
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={e => {
+            e.preventDefault();
+            handlePageChange(currentPage - 1);
+          }}
+        >
+          «
+        </a>
+      </li>
+
+      {Array.from({ length: totalPages }, (_, index) => (
+        <li
+          key={index + 1}
+          className={classNames('page-item', {
+            active: currentPage === index + 1,
+          })}
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${index + 1}`}
+            onClick={e => {
+              e.preventDefault();
+              handlePageChange(index + 1);
+            }}
+          >
+            {index + 1}
+          </a>
+        </li>
```

Comment:
When rendering a list of elements based on an array, it's important to provide a unique and stable `key` prop to each element to help React identify which items have changed, are added, or are removed. Here, the `key` prop is based on the index, which can lead to issues if the pagination pages can change dynamically (for example, if items can be added or removed, changing the total number of pages). If the pages are static, using the index is acceptable; otherwise, consider using a different attribute for the key that uniquely identifies each page link.